install/installer/scripts/kots-install.sh

```diff
@@ -0,0 +1,345 @@
+fi
+
+echo "Gitpod: auto-detecting containerd location on host machine"
+if [ -d "/mnt/node0${CONTAINERD_DIR_K3S}" ]; then
+    echo "Gitpod: containerd dir detected as k3s"
+
+    yq e -i ".workspace.runtime.containerdRuntimeDir = \"${CONTAINERD_DIR_K3S}\"" "${CONFIG_FILE}"
+elif [ -d "/mnt/node0${CONTAINERD_DIR_AL}" ]; then
+    echo "Gitpod: containerd dir detected as ${CONTAINERD_DIR_AL}"
+
+    yq e -i ".workspace.runtime.containerdRuntimeDir = \"${CONTAINERD_DIR_AL}\"" "${CONFIG_FILE}"
+fi
+
+if [ -S "/mnt/node0${CONTAINERD_SOCKET_K3S}" ]; then
+    echo "Gitpod: containerd socket detected as k3s"
+
+    yq e -i ".workspace.runtime.containerdSocket = \"${CONTAINERD_SOCKET_K3S}\"" "${CONFIG_FILE}"
+elif [ -S "/mnt/node0${CONTAINERD_SOCKET_AL}" ]; then
+    echo "Gitpod: containerd socket detected as ${CONTAINERD_SOCKET_AL}"
+
+    yq e -i ".workspace.runtime.containerdSocket = \"${CONTAINERD_SOCKET_AL}\"" "${CONFIG_FILE}"
+fi
+
+echo "Gitpod: Inject the Replicated variables into the config"
+yq e -i ".domain = \"${DOMAIN}\"" "${CONFIG_FILE}"
+yq e -i '.license.kind = "secret"' "${CONFIG_FILE}"
+yq e -i '.license.name = "gitpod-license"' "${CONFIG_FILE}"
+
+if [ "${OPEN_VSX_URL}" != "" ];
+then
+    echo "Gitpod: Setting Open VSX Registry URL"
+    yq e -i ".openVSX.url = \"${OPEN_VSX_URL}\"" "${CONFIG_FILE}"
+fi
+
+if [ "${DB_INCLUSTER_ENABLED}" = "0" ] && [ "${DB_CLOUDSQL_INSTANCE}" != "" ];
+then
+    echo "Gitpod: configuring CloudSQLProxy"
+
+    yq e -i ".database.inCluster = false" "${CONFIG_FILE}"
+    yq e -i ".database.cloudSQL.instance = \"${DB_CLOUDSQL_INSTANCE}\"" "${CONFIG_FILE}"
+    yq e -i ".database.cloudSQL.serviceAccount.kind = \"secret\"" "${CONFIG_FILE}"
+    yq e -i ".database.cloudSQL.serviceAccount.name = \"cloudsql\"" "${CONFIG_FILE}"
+fi
+
+if [ "${DB_INCLUSTER_ENABLED}" = "0" ] && [ "${DB_CLOUDSQL_INSTANCE}" = "" ];
+then
+    echo "Gitpod: configuring external database"
+
+    yq e -i ".database.inCluster = false" "${CONFIG_FILE}"
+    yq e -i ".database.external.certificate.kind = \"secret\"" "${CONFIG_FILE}"
+    yq e -i ".database.external.certificate.name = \"database\"" "${CONFIG_FILE}"
+fi
+
+if [ "${HAS_LOCAL_REGISTRY}" = "true" ];
+then
+    echo "Gitpod: configuring mirrored container registry for airgapped installation"
+
+    yq e -i ".repository = \"${LOCAL_REGISTRY_ADDRESS}\"" "${CONFIG_FILE}"
+    yq e -i ".imagePullSecrets[0].kind = \"secret\"" "${CONFIG_FILE}"
+    yq e -i ".imagePullSecrets[0].name = \"${IMAGE_PULL_SECRET_NAME}\"" "${CONFIG_FILE}"
+    yq e -i '.dropImageRepo = true' "${CONFIG_FILE}"
+
+    # Add the registry to the server allowlist - keep docker.io in case it's just using the mirrored registry functionality without being airgapped
+    yq e -i ".containerRegistry.privateBaseImageAllowList += \"${LOCAL_REGISTRY_HOST}\"" "${CONFIG_FILE}"
+    yq e -i ".containerRegistry.privateBaseImageAllowList += \"docker.io\"" "${CONFIG_FILE}"
+fi
+
+if [ "${REG_CUSTOM_DOCKER_ENABLED}" = "1" ];
```

Comment:
nit: Should it be called `REG_CUSTOM_DOCKER_CONFIG_ENABLED`? Would prefer to have `CONFIG` to be more specific.

Reply:
I've renamed it to `REG_DOCKER_CONFIG_ENABLED` to that keeps consistency with what I've done elsewhere (take the KOTS config name and captialise it). I've not changed the underlying KOTS config name (it's missing a `d`) in the expectation that someone could have started using it